.github/workflows/upload-deb-images.yml

```diff
@@ -0,0 +1,56 @@
+name: Upload package creation Docker images - DEB - amd64 and i386
+on:
+  pull_request:
+    paths:
+      - 'debs/Debian/**'
+      - 'debs/build.sh'
+  push:
+    branches:
+      - master
+      - 4.5
+      - 4.4
+    paths:
+      - 'debs/Debian/*'
+      - 'debs/build.sh'
+  workflow_dispatch:
+
+jobs:
+  Upload-deb-package-building-images:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        image: [ {CONTAINER_NAME: deb_builder_amd64, DOCKERFILE_PATH: debs/Debian/amd64}, {CONTAINER_NAME: deb_builder_i386, DOCKERFILE_PATH: debs/Debian/i386}]
+      fail-fast: false
+    steps:
+      - name: Cancel previous runs
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          cancel_others: 'true'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          skip_after_successful_duplicate: 'false'
+
+      - uses: actions/checkout@v3
+
+      - name: Copy build.sh to Dockerfile path
+        run:
+          cp $GITHUB_WORKSPACE/debs/build.sh $GITHUB_WORKSPACE/${{ matrix.image.DOCKERFILE_PATH }}
```

Comment:
I think you can avoid it by indicating the path of the build.sh in docker build command

Reply:
The Dockerfile would have to be changed, as it is there where the path to `build.sh` is specified